runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -9919,11 +9913,9 @@ J9::Z::TreeEvaluator::VMmonexitEvaluator(TR::Node * node, TR::CodeGenerator * cg
    generateS390LabelInstruction(cg, TR::InstOpCode::label, node, cFlowRegionEnd, conditions);
 
    cg->stopUsingRegister(monitorReg);
-   if (objectClassReg)
-      cg->stopUsingRegister(objectClassReg);
    if (lookupOffsetReg)
       cg->stopUsingRegister(lookupOffsetReg);
-   if (tempRegister && (tempRegister != objectClassReg))
+   if (tempRegister)
```

Comment:
Are you calling stopUsingRegisters from scratch register manager [1] that would marked registers allocated by SRM dead ? 

[1]. https://github.com/eclipse/omr/blob/8b19b8082de56b067c23bafd59dd767a58d8c02f/compiler/codegen/ScratchRegisterManager.cpp#L127

Reply:
I called `reclaimScratchRegister()` for all registers that I used. there shouldn't be any active register in the srm at this point but I guess it wont hurt to call `stopUsingRegisters()` here.

Reply:
Reclaiming a scratch register won't stop using it. It will put it back in the pool, so it can be re used within same evaluator. You do need to call stop using regs through scratch register manager as it will individually call stop using regs on the register that will decrement use count. We maintain the use count on register so that a register that is unused later on can be marked dead.